src/Workleap.OpenApi.MSBuild/ValidateContractProcess.cs

```diff
@@ -93,5 +111,24 @@ internal enum CompareCodeAgainstSpecFile
         }
 
         await Task.WhenAll(installationTasks);
+        this._loggerWrapper.LogMessage("Finished installing OpenAPI dependencies.", MessageImportance.High);
+        
+        var spectralRulesetPath = await spectralRulesetTask;
+        var spectralExecutablePath = await spectralInstallerTask;
+        
+        return new DependenciesResult(spectralRulesetPath, spectralExecutablePath);
+    }
+    
+    private class DependenciesResult
+    {
```

Comment:
We define this class twice, can we reuse any logic?

Reply:
similarly for the other methods in this class, any opportunities to share any code between them?